File: src/code-here/Example_v1.js
'use strict';

function decide(s) {
  var NET_X = 216;              // GROUND_HALF_WIDTH
  var GROUND_WIDTH = 432;
  var PLAYER_HALF_LENGTH = 32;
  var towardNet = s.side === 'RIGHT' ? -1 : 1;

  var ownNearBoundary = s.side === 'RIGHT' ? NET_X : 0;
  var ownFarBoundary  = s.side === 'RIGHT' ? GROUND_WIDTH : NET_X;
  var standbyX = (ownNearBoundary + ownFarBoundary) / 2;

  var landingOnOwnSide =
    s.ball.expectedLandingPointX > ownNearBoundary &&
    s.ball.expectedLandingPointX < ownFarBoundary;

  var targetX;
  if (landingOnOwnSide) {
    // 공이 진짜 우리 쪽으로 온다 -- 낙하지점보다 살짝 "네트 반대쪽"에 서서
    // 몸의 네트쪽 면으로 공을 받는다 (=> 공이 네트 쪽으로 튀어나감).
    targetX = s.ball.expectedLandingPointX - towardNet * 12;
  } else {
    // 아직 상대 쪽에 있다 -- 뒤로 물러나 대기.
    targetX = standbyX;
  }

  var dx = targetX - s.self.x;
  var x = 0;
  if (Math.abs(dx) > 6) {
    x = dx > 0 ? 1 : -1;
  }

  // 점프 조건: 엄격하게 (윗해설 참고)
  var y = 0;
  var xAligned = Math.abs(s.ball.x - s.self.x) < PLAYER_HALF_LENGTH;
  var ballSlowSideways = Math.abs(s.ball.xVelocity) < 5;
  var ballClearlyHigh = s.ball.y < 150;
  if (
    s.self.state === 0 &&
    xAligned &&
    ballSlowSideways &&
    ballClearlyHigh &&
    s.ball.yVelocity > 0
  ) {
    y = -1;   // 점프 = 땅에서 y = -1
  }

  // hit은 항상 0 -- 파워히트도, 다이빙도 없음.
  return { x: x, y: y, hit: 0 };
}

File: src/code-here/Example_v1.py
def decide(s):
    NET_X = 216
    GROUND_WIDTH = 432
    PLAYER_HALF_LENGTH = 32
    toward_net = -1 if s['side'] == 'RIGHT' else 1

    own_near = NET_X if s['side'] == 'RIGHT' else 0
    own_far  = GROUND_WIDTH if s['side'] == 'RIGHT' else NET_X
    standby_x = (own_near + own_far) / 2

    landing_on_own_side = (
        s['ball']['expectedLandingPointX'] > own_near
        and s['ball']['expectedLandingPointX'] < own_far
    )

    if landing_on_own_side:
        target_x = s['ball']['expectedLandingPointX'] - toward_net * 12
    else:
        target_x = standby_x

    dx = target_x - s['self']['x']
    x = 0
    if abs(dx) > 6:
        x = 1 if dx > 0 else -1

    y = 0
    x_aligned = abs(s['ball']['x'] - s['self']['x']) < PLAYER_HALF_LENGTH
    ball_slow_sideways = abs(s['ball']['xVelocity']) < 5
    ball_clearly_high = s['ball']['y'] < 150
    if (
        s['self']['state'] == 0
        and x_aligned
        and ball_slow_sideways
        and ball_clearly_high
        and s['ball']['yVelocity'] > 0
    ):
        y = -1

    return {'x': x, 'y': y, 'hit': 0}